### PR TITLE
feat: upgrade react@19 and next@15

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

All started from the Vercel costs issue.
I've found out that next@14 has issues with caching with causes the ISR Writes issues:
https://www.reddit.com/r/nextjs/comments/1gdguci/isr_writes_insanely_high/

1. Updated `next` to 15 using [codemod](https://nextjs.org/docs/app/building-your-application/upgrading/version-15)
2. Updated `react` to 19 using [codemod](https://react.dev/blog/2024/04/25/react-19-upgrade-guide)

# To Test

Please make a sanity check of CoW Swap, Explorer and landings.
No need to test deeply, just go through all pages.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -182,27 +183,27 @@
...
-    "react-dom": "18.2.0",
+    "react-dom": "19.0.0-rc-66855b96-20241106",
     "react-feather": "^2.0.10",
     "react-ga4": "^1.4.1",
     "react-helmet": "^6.1.0",
     "react-icons": "^5.2.1",
     "react-inlinesvg": "^3.0.1",
     "react-intersection-observer": "^9.10.1",
-    "react-is": "18.2.0",
+    "react-is": "19.0.0-rc-66855b96-20241106",
```
Comment (shoom3301) [inline package.json:206]: This version was set automatically by [codemod](https://react.dev/blog/2024/04/25/react-19-upgrade-guide)

File: package.json [hunk 2]
```diff
@@ -275,7 +275,7 @@
     "@types/node": "18.16.9",
     "@types/react-copy-to-clipboard": "^4.3.0",
     "@types/react-cytoscapejs": "^1.2.2",
-    "@types/react-dom": "18.2.14",
+    "@types/react-dom": "npm:types-react-dom@19.0.0-rc.1",
```
Comment (anxolin) [inline package.json:278]: this feals like a strange change to me...

**Review by anxolin — APPROVED** [review]
Amazing! what is this witchery of codemod! did it really work out of the box?

**Comment by shoom3301** [comment]
Yes, I just ran the script and that is it ✨

File: package.json (hunk 2 above)
Comment (shoom3301) [reply package.json:278]: Me too, codemod did this

**Comment by shoom3301** [comment]
@elena-zh thank you!
Fixed

**Comment by elena-zh** [comment]
Hey @shoom3301 , batch [graph ](https://explorer-dev-git-feat-upgrade-core-deps-cowswap.vercel.app/tx/0x60806d4176ebb3806ac234e1d6f29822523c06f9cb9e9ea4b7633156b3145717/?tab=graph&vis=transfers)page is still crashing 
image

**Review by elena-zh — APPROVED** [review]
Works fine now!
The issue with 'reading h' is reported in #3636

**Pull request merged**